src/test.js

```diff
@@ -251,17 +249,18 @@ Test.prototype = {
 
   queueHook (hook, hookName, hookOwner) {
     const callHook = () => {
-      const promise = hook.call(this.testEnvironment, this.assert);
+      let promise;
+      if (hookName === 'before' || hookName === 'after') {
+        promise = hook.call(this.module.testEnvironment, this.assert);
```

Comment:
there's still an issue here which is that the `before` and `after` hooks can be called with the wrong module's testEnvironment depending on the ordering of the tests. Instead of calling the hook with the _current test_'s module, it should be called with the module the hook belongs to.

Reply:
In https://github.com/qunitjs/qunit/pull/1762/commits/759a391c40fbe8bdab391ba50c59ce76a5f9b912 I update to using `hookOwner` instead of `this.module` which resolves this problem.